packages/Search/examples/bvh_driver/bvh_driver.cpp

```diff
@@ -95,110 +46,97 @@ int main_( Teuchos::CommandLineProcessor &clp, int argc, char *argv[] )
+                              double const y = random_points( i )[1];
+                              double const z = random_points( i )[2];
+                              bounding_boxes( i ) = {
+                                  {{x - 1., y - 1., z - 1.}},
+                                  {{x + 1., y + 1., z + 1.}}};
+                          } );
+    Kokkos::fence();
+
+    std::ostream &os = std::cout;
+
+    auto start = std::chrono::high_resolution_clock::now();
     DataTransferKit::BVH<DeviceType> bvh( bounding_boxes );
+    auto end = std::chrono::high_resolution_clock::now();
+    std::chrono::duration<double> elapsed_seconds = end - start;
+    os << "construction " << elapsed_seconds.count() << "\n";
 
-    // random points for radius search and kNN queries
-    auto queries = make_random_cloud( Lx, Ly, Lz, n_points );
-    Kokkos::View<double * [3], ExecutionSpace> point_coords( "point_coords",
-                                                             n_points );
-
-    auto point_coords_host = Kokkos::create_mirror_view( point_coords );
-    for ( int i = 0; i < n_points; ++i )
-    {
-        auto const &point = queries[i];
-        point_coords_host( i, 0 ) = std::get<0>( point );
-        point_coords_host( i, 1 ) = std::get<1>( point );
-        point_coords_host( i, 2 ) = std::get<2>( point );
-    }
-    Kokkos::deep_copy( point_coords, point_coords_host );
-
-    std::default_random_engine generator;
-
-    if ( mode == "knn" )
     {
-        Kokkos::View<int *, ExecutionSpace> k( "distribution_k", n_points );
-        auto k_host = Kokkos::create_mirror_view( k );
-
-        // use random number k of for the kNN search
-        int max_k = std::floor( sqrt( nx * nx + ny * ny + nz * nz ) );
-        std::uniform_int_distribution<int> distribution_k( 1, max_k );
-        for ( int i = 0; i < n_points; ++i )
-        {
-            k_host[i] = distribution_k( generator );
-        }
-        Kokkos::deep_copy( k, k_host );
-
-        Kokkos::View<details::Nearest<DataTransferKit::Point> *, DeviceType>
-            nearest_queries( "nearest_queries", n_points );
+        Kokkos::View<
+            DataTransferKit::Details::Nearest<DataTransferKit::Point> *,
+            DeviceType>
+            queries( "queries", n_queries );
         Kokkos::parallel_for(
-            DTK_MARK_REGION( "register_nearest_queries" ),
-            Kokkos::RangePolicy<ExecutionSpace>( 0, n_points ),
+            Kokkos::RangePolicy<ExecutionSpace>( 0, n_queries ),
             KOKKOS_LAMBDA( int i ) {
-                nearest_queries( i ) = details::nearest<DataTransferKit::Point>(
-                    {{point_coords( i, 0 ), point_coords( i, 1 ),
-                      point_coords( i, 2 )}},
-                    k( i ) );
+                queries( i ) =
+                    DataTransferKit::Details::nearest<DataTransferKit::Point>(
+                        random_points( i ), n_neighbors );
```

Comment:
So your target points coincide with your source points? Was that done for simplicity?